pvnet/models/multimodal/validation/config_validation.py

```
@@ -88,105 +101,161 @@ def validate(
+        logger.debug(f"'{key}' shape validation passed (time resolution: {time_res_mins} mins).")
 
-    # Solar
     config_key_sun = "include_sun"
     if cfg.get(config_key_sun):
         logger.debug("Validating modality: sun (azimuth, elevation)")
-        possible_fallback_keys = ["gsp", "site", "pv"]
-        fallback_key = next(
-            (k for k in possible_fallback_keys if k in input_data_config and
-                isinstance(input_data_config.get(k), dict)),
-            None
-        )
-        if fallback_key is None:
-            raise KeyError(
-                 "Cannot determine interval for sun: No suitable fallback modality "
-                 "('gsp', 'site', or 'pv') found as dictionary in input_data_config."
-             )
-        sun_interval = get_modality_interval(input_data_config, "sun", fallback_key)
+        sun_time_res_mins = None
+        try:
+            sun_time_res_mins = get_modality_interval(input_data_dict, "sun", None)
+            logger.debug(
+                f"Using direct 'sun' configuration for time resolution: "
+                f"{sun_time_res_mins} mins."
+            )
+        except (KeyError, ValueError) as e_sun_direct:
+            logger.debug(
+                f"Direct 'sun' config for interval not found or invalid. "
+                f"Attempting fallback: {e_sun_direct}"
+            )
+            possible_fallback_keys = ["gsp", "site", "pv"]
+            fallback_key_sun = next(
+                (k_sun for k_sun in possible_fallback_keys if k_sun in input_data_dict and
+                    isinstance(input_data_dict.get(k_sun), dict)),
+                None
+            )
+            if fallback_key_sun is not None:
+                try:
+                    sun_time_res_mins = get_modality_interval(
+                        input_data_dict, "sun", fallback_key_sun
+                    )
+                    logger.debug(
```

I feel like if there is already fallback modality handling within get_modality_interval we can just put sun + backup key through it right away and let it either handle it and raise appropriate warnings or fail. Is there a reason for these try/excepts I'm not seeing?

---

Also, strictly speaking, I think if sun_posiotion is not explicitly in data config it will not be in the batches either? At least for dsampler-created batches, because I think currently we expect sun position to be added on creation and only if sun pos config is present. We used to have this situation in datapipes where sun position would be calculated by them after batch creation, so you wouldn't have this data in the batch initially and it would be added later on, but we don't do that anymore.

Is there some mid-way models we are trying to accommodate for here? If not then I would say we should just fail hard if there is include_sun in config and no sun data/there is sun data but no data config, because that just means it's not the batch's config.